### DISCUSSION TITLE
关于Select value不存在于option项

### DISCUSSION THREAD

**Discussion by 2449847346** [ant-design/ant-design#31366] [Q&A]
react Select，如果value不存在于Select中的option选项中，有没有什么办法让他不展示啊，现在每次form.setFieldValues之前每次都需要把option数据获取到然后再循环看是否存在，如果存跳过，不存在将该字段值变为undefined，在angular Select中发现默认带这种功能

**Comment by yoyo837** [comment]
个人觉得就应该展示，你应该检查为什么option里没有预期选项的问题

**Comment by 2449847346** [comment]
这些option是被创建的，有被删除的可能性，被删除后，老数据储存就会有问题，会展示相对应的value，这时候用户体验可能不会好，我是在用react重构angular的项目，angular antd 可以自动忽略option无对应value，但react antd不行，老项目又会有遗留数据，我方这边又硬要不展示，每个东西都需要异步获取option再过滤，觉得还是有需要出个对应api，你想展示可展示，不想展示也可以过滤掉无对应value；个人见解吧；




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年7月13日(星期二) 上午9:48
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ant-design/ant-design] 关于Select value不存在于option项 (#31366)





 
个人觉得就应该展示，你应该检查为什么option里没有预期选项的问题

**Reply by yoyo837** [reply]
目前确实好像不支持，要不，你先抽取个工具函数，或者把组件二次包装一下复用。

**Comment by 2449847346** [comment]
好的，我先看看



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年7月13日(星期二) 上午10:25
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ant-design/ant-design] 关于Select value不存在于option项 (#31366)





 
目前确实好像不支持，要不，你先抽取个工具函数，或者把组件二次包装一下复用。
